app/src/main/java/com/thailam/weatherwhen/MainActivity.kt

```diff
@@ -1,12 +1,84 @@
 package com.thailam.weatherwhen
 
+import android.annotation.SuppressLint
+import android.content.pm.PackageManager
+import android.location.Location
 import android.os.Bundle
+import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.Observer
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationServices.getFusedLocationProviderClient
+import com.thailam.weatherwhen.viewmodel.ForecastViewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class MainActivity : AppCompatActivity() {
 
+    private val forecastViewModel: ForecastViewModel by viewModel()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        initViewModel()
+        checkPermissions()
+    }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
+        if (requestCode != PERMISSION_CODE) return
+        for (x in 0 until permissions.size) {
+            if (isPermissionGranted(PERMISSION_FINE_LOCATION)) getLastLocation()
```

Comment:
anh nhớ là check if permissions.size và grantResults[0] mà, nếu viết vòng for để loại bỏ trường hợp đó thì có vẻ khó đọc quá